### DISCUSSION TITLE
How to stop service onInit hook from running when testing

### DISCUSSION THREAD

**Discussion by janek515** [TypedProject/ts-express-decorators#2147] [Q&A] [answered]
Hi, 
I have a simple setup with a service and a controller. The service has an onInit hook that connects to a DB. I'm trying to make a unit test for the controller. I would like to mock that service and not have its onInit hook called. I know how to mock it, but it appears that the onInit hook is ran by default when i call `beforeAll(PlatformTest.bootstrap(Server));`. Is there a simple solution? I couldn't find any info in the docs.
Thanks in advance.

**Comment by Romakita** [comment, marked as answer]
Hello @janek515 

The $onInit is always emitted and changing that is not recommended.

The simple solution if to set an option to by-pass connection during the bootstrap sequence:

```typescript
beforeAll(PlatformTest.bootstrap(Server, {
   mydb: {
      disabled: true
   }
}));

@Injectable()
class MyConnectionService {
    @Constant("mydb.disabled", false)
    disabled: boolean;

   $onInit() {
     if (!this.disabled) {
        ///....
     }
   }
}
```
